src/RemoteInboxNotifications/StoredStateSetupForProducts.php

```diff
@@ -112,19 +108,23 @@ public static function run_on_transition_post_status( $new_status, $old_status,
 			return;
 		}
 
-		$stored_state                         = RemoteInboxNotificationsEngine::get_stored_state();
-		$stored_state->there_are_now_products = true;
-
-		RemoteInboxNotificationsEngine::update_stored_state( $stored_state );
-
-		self::enqueue_async_run_remote_notifications();
+		self::update_stored_state_and_run_remote_notifications();
 	}
 
 	/**
 	 * Enqueues an async action (using action-scheduler) to run remote
 	 * notifications.
 	 */
-	private static function enqueue_async_run_remote_notifications() {
+	private static function update_stored_state_and_run_remote_notifications() {
```

Comment:
Since we're conditionally running the remote notification, I think it's better to name it something like 
```suggestion
	private static function update_stored_state_and_possibly_run_remote_notifications() {
```